### DISCUSSION TITLE
Ideas, Questions, Anything

### DISCUSSION THREAD

**Discussion by mechatroner** [mechatroner/vscode_rainbow_csv#79] [General]
Hello, Everyone!

**Comment by mkennedm** [comment]
How do you use `IN` expressions? For the example at the bottom of https://rbql.org/ I can search for all movies released in 2012 with `select * where a['Year'] == 2012`.

I can also search for all movies released in 2013 with `select * where a['Year'] == 2013`.

If I would like to search for all movies released in either 2012 or 2013 I can do `select * where a['Year'] == 2012 || a['Year'] == 2013`.

How do I do the above query using the `IN` operator?

`select * where a['Year'] in (2012, 2013)`
returns an error
"At record 1, Details: Cannot use 'in' operator to search for '2012' in 2013"

`select * where a['Year'] in [2012, 2013]`
returns an empty table.

How do I search a table for values that appear in a list?

**Reply by mechatroner** [reply]
This happens because RBQL is a thin wrapper around JS, and the `in` operator in JavaScript works in a slightly different way than in Python (or SQL): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/in_operator_no_object - It basically check if the object on the right side has a key which is on the left side of the `in` operator. So to fix this you can either use this form:
```
select * where ['2012' ,'2013'].indexOf(a['Year']) != -1
```
or
```
select * where a['Year'] in {2012: 1, 2013: 1}
```
Also when you use `indexOf` you will have to either put strings into the array or convert the year to an integer, otherwise, it will not find anything, because there is no automatic type conversions as with some other JS operators.

Thanks for the question, BTW! I learned that JS has `in` operator - never used it myself, because I prefer to always check keys with `hasOwnProperty()`.

**Comment by denyspro** [comment]
Hey, is there a way to query the error lines (the ones that didn't pass the CSVLint check)? So we can review them all at once?

It would be cool if the CSVLint went to the line/row with the error when you click on it.

**Reply by ballendq** [reply]
Would love this feature too.

**Reply by ballendq** [reply]
For some of the data i've been working with, a useful hack is to select the "Align" option. Selecting this option will say which line has an error. After "Align" stops complaining, try CSVLint again. CSVLint should also pass.
